### ISSUE TITLE
Check if problem-json is content type for error status codes

### ISSUE THREAD

**Issue by fmueller** [zalando-incubator/zally#28]
If https://github.com/swagger-api/swagger-parser/pull/303/files is merged, we can check actual content type.

**Comment by netme** [comment]
The issue was closed with:

```
Hi, we don't support this in the spec, so have to punt in the parser :( Consider adding an extension to do this, and I'd be happy to add it to the repo as well.
```

Most probably we need a plan B :-)

**Comment by netme** [comment]
Did few requests against latest master:

```bash
curl -i -X GET http://localhost:8080/
HTTP/1.1 405 
X-Application-Context: application:dev
Allow: POST
Content-Type: application/problem+json
Transfer-Encoding: chunked
Date: Thu, 09 Mar 2017 21:41:15 GMT

{"title":"Method Not Allowed","status":405,"detail":"Request method 'GET' not supported"}
```

```bash
curl -i -X POST http://localhost:8080/
HTTP/1.1 400 
X-Application-Context: application:dev
Content-Type: application/problem+json
Transfer-Encoding: chunked
Date: Thu, 09 Mar 2017 21:42:12 GMT
Connection: close

{"title":"Bad Request","status":400,"detail":"Required request body is missing: public org.springframework.http.ResponseEntity<com.fasterxml.jackson.databind.JsonNode> de.zalando.zally.ApiViolationsController.validate(com.fasterxml.jackson.databind.JsonNode)"}
```

```bash
curl -i -X POST -H "Content-Type: application/json" http://localhost:8080/ -d '{"abcde": 123'
HTTP/1.1 400 
X-Application-Context: application:dev
Content-Type: application/problem+json
Transfer-Encoding: chunked
Date: Thu, 09 Mar 2017 21:43:30 GMT
Connection: close

{"title":"Bad Request","status":400,"detail":"Could not read document: Unexpected end-of-input: expected close marker for Object (start marker at [Source: java.io.PushbackInputStream@4a44189f; line: 1, column: 1])\n at [Source: java.io.PushbackInputStream@4a44189f; line: 1, column: 40]; nested exception is com.fasterxml.jackson.core.io.JsonEOFException: Unexpected end-of-input: expected close marker for Object (start marker at [Source: java.io.PushbackInputStream@4a44189f; line: 1, column: 1])\n at [Source: java.io.PushbackInputStream@4a44189f; line: 1, column: 40]"
```

It's working as expected. Closing this issue.

**Comment by netme** [comment]
Sorry, my misunderstanding. Reopening.

**Comment by netme** [comment]
I've tried to find other PRs to swagger repos from this guy, no chance.

@fmueller should we postpone it and move out of the scope of MS 1.0?

**Comment by fmueller** [comment]
@netme Good point, I don't think it makes sense to keep it in the scope of version 1.0. We'll postpone it.

Thanks for following up ;)

**Comment by netme** [comment]
Implemented in #426
